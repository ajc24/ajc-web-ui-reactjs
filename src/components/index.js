/**
 * Developed by Anthony Cox in 2024
 */

/* Export the header components */
export { default as HeaderTypeA } from './header/HeaderTypeA';

/* Export the main components */
export { default as MainContent } from './main/MainContent';

/* Export the menu bar components */
export { default as MenuBarTypeA } from './menu-bar/MenuBarTypeA';
