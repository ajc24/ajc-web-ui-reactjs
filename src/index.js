/**
 * Developed by Anthony Cox in 2024
 */

/* Import the header components */
import Header from './components/header/Header';

/* Import the image components */
import DecorativeImage from './components/images/DecorativeImage';
import FunctionalImage from './components/images/FunctionalImage';
import InformativeImage from './components/images/InformativeImage';

/* Export all components */
export {
  DecorativeImage,
  FunctionalImage,
  InformativeImage,
  Header,
};
