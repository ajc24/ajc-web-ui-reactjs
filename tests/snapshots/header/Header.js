/**
 * Developed by Anthony Cox in 2025
 */
import 'jsdom-global/register';
import React from 'react';
import { TestDev } from 'ajc-testing-code';
import { Header } from '../../../src';
import storybookLogo from '../../../stories/images/files/storybook-logo.svg';
import storybookLogoLong from '../../../stories/images/files/storybook-logo-long.svg';

describe('Header', () => {
  /* Set the IDs for use in the tests */
  const testComponentId = 'test-header-id';

  describe('Default (small) Header, with custom content, no logo image, no title text, no subtitle text', () => {
    let snapshot;
    
    beforeAll(() => {
      /* Generate the snapshot for the component */
      snapshot = TestDev.createSnapshot(
        <React.Fragment>
          <Header id={testComponentId}>
            Header component text content.
          </Header>
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('Small Header, with square logo image, no title text, no subtitle text', () => {
    let snapshot;
    
    beforeAll(() => {
      /* Generate the snapshot for the component */
      snapshot = TestDev.createSnapshot_UseFakeTimers(
        <React.Fragment>
          <Header id={testComponentId} logoSrc={storybookLogo} logoType="square" size="small" />
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('Small Header, with rectangle logo image, no title text, no subtitle text', () => {
    let snapshot;
    
    beforeAll(() => {
      /* Generate the snapshot for the component */
      snapshot = TestDev.createSnapshot_UseFakeTimers(
        <React.Fragment>
          <Header id={testComponentId} logoSrc={storybookLogoLong} logoType="rectangle" size="small" />
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('Tall Header, with logo image, no title text, no subtitle text', () => {
    let snapshot;
    
    beforeAll(() => {
      /* Generate the snapshot for the component */
      snapshot = TestDev.createSnapshot_UseFakeTimers(
        <React.Fragment>
          <Header id={testComponentId} logoSrc={storybookLogoLong} size="tall" />
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('Small Header, no logo image, with title text, no subtitle text', () => {
    let snapshot;
    
    beforeAll(() => {
      /* Generate the snapshot for the component */
      snapshot = TestDev.createSnapshot_UseFakeTimers(
        <React.Fragment>
          <Header id={testComponentId} size="small" titleTextColour="black" titleTextContent="Header Title Text Content" />
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('Small Header, no logo image, with title text, with subtitle text', () => {
    let snapshot;
    
    beforeAll(() => {
      /* Generate the snapshot for the component */
      snapshot = TestDev.createSnapshot_UseFakeTimers(
        <React.Fragment>
          <Header id={testComponentId} size="small" subtitleTextContent="Subtitle text content" titleTextContent="Header Title Text Content" />
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('Tall Header, no logo image, with title text, no subtitle text', () => {
    let snapshot;
    
    beforeAll(() => {
      /* Generate the snapshot for the component */
      snapshot = TestDev.createSnapshot_UseFakeTimers(
        <React.Fragment>
          <Header id={testComponentId} size="tall" titleTextColour="black" titleTextContent="Header Title Text Content" />
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('Tall Header, no logo image, with title text, with subtitle text', () => {
    let snapshot;
    
    beforeAll(() => {
      /* Generate the snapshot for the component */
      snapshot = TestDev.createSnapshot_UseFakeTimers(
        <React.Fragment>
          <Header id={testComponentId} size="tall" subtitleTextContent="Subtitle text content" titleTextContent="Header Title Text Content" />
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });
});
