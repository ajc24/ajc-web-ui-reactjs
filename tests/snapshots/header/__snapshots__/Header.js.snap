// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Default (small) Header, with custom content, no logo image, no title text, no subtitle text verifies the snapshot for the component 1`] = `"<header role="banner" id="test-header-id--header-base" class="header screen-width-root background-white font-default font-black"><div class="header-content-outer screen-width-content-outer background-white" data-bgimage="false"><div class="header-content-inner screen-width-content-inner header-small-height-no-border">Header component text content.</div></div></header>"`;

exports[`Header Small Header, no logo image, with title text, no subtitle text verifies the snapshot for the component 1`] = `"<header role="banner" id="test-header-id--header-base" class="header screen-width-root background-white font-default font-black"><div class="header-content-outer screen-width-content-outer background-white" data-bgimage="false"><div class="header-content-inner screen-width-content-inner header-small-height-no-border"><div class="header-title-subtitle-text-container header-title-subtitle-text-container-width-no-logo"><div id="test-header-id--header-title-text" class="background-transparent title-text-size-inside-header title-text-alignment-centre" data-wrap="false"><h1 class="title-text-default title-text-font-black" aria-label="Header Title Text Content" style="font-size: 3rem; white-space: nowrap;">Header Title Text Content</h1></div></div></div></div></header>"`;

exports[`Header Small Header, no logo image, with title text, with subtitle text verifies the snapshot for the component 1`] = `"<header role="banner" id="test-header-id--header-base" class="header screen-width-root background-white font-default font-black"><div class="header-content-outer screen-width-content-outer background-white" data-bgimage="false"><div class="header-content-inner screen-width-content-inner header-small-height-no-border"><div class="header-title-subtitle-text-container header-title-subtitle-text-container-width-no-logo"><div id="test-header-id--header-title-text" class="background-transparent title-text-size-inside-header title-text-alignment-centre" data-wrap="false"><h1 class="title-text-default title-text-font-black" aria-label="Header Title Text Content" style="font-size: 3rem; white-space: nowrap;">Header Title Text Content</h1></div><div id="test-header-id--header-subtitle-text" class="background-transparent subtitle-text-size-inside-header subtitle-text-alignment-centre subtitle-text-with-header-text"><p class="subtitle-text-default font-black" aria-hidden="false">Subtitle text content</p></div></div></div></div></header>"`;

exports[`Header Small Header, with logo image, with title text, no subtitle text verifies the snapshot for the component 1`] = `"<header role="banner" id="test-header-id--header-base" class="header screen-width-root background-white font-default font-black"><div class="header-content-outer screen-width-content-outer background-white" data-bgimage="false"><div class="header-content-inner screen-width-content-inner header-small-height-no-border"><div class="header-small-logo-title-subtitle-text-container"><div class="background-transparent image-header-logo-inside-header image-header-logo-alignment-left"><img id="test-header-id--header-logo--decorative-image" class="image image-animations" width="140" height="144" alt="" src="[object Object]"></div><div class="header-title-subtitle-text-container header-title-subtitle-text-container-width-with-logo"><div id="test-header-id--header-title-text" class="background-transparent title-text-size-inside-header title-text-alignment-left" data-wrap="false"><h1 class="title-text-default title-text-font-black" aria-label="Header Title Text Content" style="font-size: 3rem; white-space: nowrap;">Header Title Text Content</h1></div></div></div></div></div></header>"`;

exports[`Header Small Header, with logo image, with title text, with subtitle text verifies the snapshot for the component 1`] = `"<header role="banner" id="test-header-id--header-base" class="header screen-width-root background-white font-default font-black"><div class="header-content-outer screen-width-content-outer background-white" data-bgimage="false"><div class="header-content-inner screen-width-content-inner header-small-height-no-border"><div class="header-small-logo-title-subtitle-text-container"><div class="background-transparent image-header-logo-inside-header image-header-logo-alignment-left"><img id="test-header-id--header-logo--decorative-image" class="image image-animations" width="140" height="144" alt="" src="[object Object]"></div><div class="header-title-subtitle-text-container header-title-subtitle-text-container-width-with-logo"><div id="test-header-id--header-title-text" class="background-transparent title-text-size-inside-header title-text-alignment-left" data-wrap="false"><h1 class="title-text-default title-text-font-black" aria-label="Header Title Text Content" style="font-size: 3rem; white-space: nowrap;">Header Title Text Content</h1></div><div id="test-header-id--header-subtitle-text" class="background-transparent subtitle-text-size-inside-header subtitle-text-alignment-left subtitle-text-with-header-text"><p class="subtitle-text-default font-black" aria-hidden="false">Subtitle text content</p></div></div></div></div></div></header>"`;

exports[`Header Small Header, with rectangle logo image, no title text, no subtitle text verifies the snapshot for the component 1`] = `"<header role="banner" id="test-header-id--header-base" class="header screen-width-root background-white font-default font-black"><div class="header-content-outer screen-width-content-outer background-white" data-bgimage="false"><div class="header-content-inner screen-width-content-inner header-small-height-no-border"><div class="background-transparent image-header-logo-inside-header image-header-logo-alignment-centre"><img id="test-header-id--header-logo--informative-image" class="image image-animations" width="312" height="144" alt="No alternate text has been specified for this informative image." src="[object Object]"></div></div></div></header>"`;

exports[`Header Small Header, with square logo image, no title text, no subtitle text verifies the snapshot for the component 1`] = `"<header role="banner" id="test-header-id--header-base" class="header screen-width-root background-white font-default font-black"><div class="header-content-outer screen-width-content-outer background-white" data-bgimage="false"><div class="header-content-inner screen-width-content-inner header-small-height-no-border"><div class="background-transparent image-header-logo-inside-header image-header-logo-alignment-centre"><img id="test-header-id--header-logo--informative-image" class="image image-animations" width="140" height="144" alt="No alternate text has been specified for this informative image." src="[object Object]"></div></div></div></header>"`;

exports[`Header Tall Header, no logo image, with title text, no subtitle text verifies the snapshot for the component 1`] = `"<header role="banner" id="test-header-id--header-base" class="header screen-width-root background-white font-default font-black"><div class="header-content-outer screen-width-content-outer background-white" data-bgimage="false"><div class="header-content-inner screen-width-content-inner header-tall-height-no-border"><div class="header-title-subtitle-text-container"><div id="test-header-id--header-title-text" class="background-transparent title-text-size-inside-header title-text-alignment-centre" data-wrap="false"><h1 class="title-text-default title-text-font-black" aria-label="Header Title Text Content" style="font-size: 3rem; white-space: nowrap;">Header Title Text Content</h1></div></div></div></div></header>"`;

exports[`Header Tall Header, no logo image, with title text, with subtitle text verifies the snapshot for the component 1`] = `"<header role="banner" id="test-header-id--header-base" class="header screen-width-root background-white font-default font-black"><div class="header-content-outer screen-width-content-outer background-white" data-bgimage="false"><div class="header-content-inner screen-width-content-inner header-tall-height-no-border"><div class="header-title-subtitle-text-container"><div id="test-header-id--header-title-text" class="background-transparent title-text-size-inside-header title-text-alignment-centre" data-wrap="false"><h1 class="title-text-default title-text-font-black" aria-label="Header Title Text Content" style="font-size: 3rem; white-space: nowrap;">Header Title Text Content</h1></div><div id="test-header-id--header-subtitle-text" class="background-transparent subtitle-text-size-inside-header subtitle-text-alignment-centre subtitle-text-with-header-text"><p class="subtitle-text-default font-black" aria-hidden="false">Subtitle text content</p></div></div></div></div></header>"`;

exports[`Header Tall Header, with logo image, no title text, no subtitle text verifies the snapshot for the component 1`] = `"<header role="banner" id="test-header-id--header-base" class="header screen-width-root background-white font-default font-black"><div class="header-content-outer screen-width-content-outer background-white" data-bgimage="false"><div class="header-content-inner screen-width-content-inner header-tall-height-no-border"><div class="background-transparent image-header-logo-inside-header image-header-logo-alignment-centre"><img id="test-header-id--header-logo--informative-image" class="image image-animations" width="312" height="304" alt="No alternate text has been specified for this informative image." src="[object Object]"></div></div></div></header>"`;

exports[`Header Tall Header, with logo image, with title text, no subtitle text verifies the snapshot for the component 1`] = `"<header role="banner" id="test-header-id--header-base" class="header screen-width-root background-white font-default font-black"><div class="header-content-outer screen-width-content-outer background-white" data-bgimage="false"><div class="header-content-inner screen-width-content-inner header-tall-height-no-border"><div class="header-title-subtitle-text-container"><div class="background-transparent image-header-logo-inside-header image-header-logo-alignment-centre"><img id="test-header-id--header-logo--decorative-image" class="image image-animations" width="312" height="304" alt="" src="[object Object]"></div><div id="test-header-id--header-title-text" class="background-transparent title-text-size-inside-header title-text-alignment-centre" data-wrap="false"><h1 class="title-text-default title-text-font-black" aria-label="Header Title Text Content" style="font-size: 3rem; white-space: nowrap;">Header Title Text Content</h1></div></div></div></div></header>"`;

exports[`Header Tall Header, with logo image, with title text, with subtitle text verifies the snapshot for the component 1`] = `"<header role="banner" id="test-header-id--header-base" class="header screen-width-root background-white font-default font-black"><div class="header-content-outer screen-width-content-outer background-white" data-bgimage="false"><div class="header-content-inner screen-width-content-inner header-tall-height-no-border"><div class="header-title-subtitle-text-container"><div class="background-transparent image-header-logo-inside-header image-header-logo-alignment-centre"><img id="test-header-id--header-logo--decorative-image" class="image image-animations" width="312" height="304" alt="" src="[object Object]"></div><div id="test-header-id--header-title-text" class="background-transparent title-text-size-inside-header title-text-alignment-centre" data-wrap="false"><h1 class="title-text-default title-text-font-black" aria-label="Header Title Text Content" style="font-size: 3rem; white-space: nowrap;">Header Title Text Content</h1></div><div id="test-header-id--header-subtitle-text" class="background-transparent subtitle-text-size-inside-header subtitle-text-alignment-centre subtitle-text-with-header-text"><p class="subtitle-text-default font-black" aria-hidden="false">Subtitle text content</p></div></div></div></div></header>"`;
